src/android/InAppBrowserDriver.java

```
@@ -0,0 +1,543 @@
+        }
+
+        return null;
+    }
+
+    /**
+     * Notify the host application that there was an error loading the page.
+     *
+     * @param errorCode     The error code received.
+     * @param description   Description of the error.
+     * @param url           The url the error was received for.
+     *
+     * @return              Object to stop propagation or null
+     */
+    public Object onReceivedError(int errorCode, String description, String url) {
+        try {
+            JSONObject obj = new JSONObject();
+            obj.put("type", InAppBrowser.LOAD_ERROR_EVENT);
+            obj.put("url", url);
+            obj.put("code", errorCode);
+            obj.put("message", description);
+
+            inAppBrowser.sendUpdate(obj, true, PluginResult.Status.ERROR);
+        } catch (JSONException ex) {
+            Log.d(LOG_TAG, "Should never happen");
+        }
+
+        // prevent the default behavior
+        return new Object();
+    }
+
+    /**
+     * Notify the host application that there was a HTTP auth challenge issued.
+     *
+     * @param view      The WebView that is initiating the callback
+     * @param handler   The HttpAuthHandler used to set the WebView's response
+     * @param host      The host requiring authentication
+     * @param realm     The realm for which authentication is required
+     *
+     * @return          Returns true if host application will resolve this auth challenge, otherwise false
+     */
+    public Boolean onReceivedHttpAuthRequest(CordovaWebView view, ICordovaHttpAuthHandler handler, String host, String realm) {
+        // Check if there is some plugin which can resolve this auth challenge
+        PluginManager pluginManager = null;
+        try {
+            pluginManager = (PluginManager)InAppBrowserImpl.invokeOn(inAppBrowser.getWebView().getView(), "getPluginManager", new Class[0], new Object[0]);
+        } catch (InAppBrowserImpl.InvokeException e) {
```

Could we put a log or something in this empty catch?